packages/sources/coinmarketcap-test/src/crypto-utils.ts

```diff
@@ -0,0 +1,188 @@
+import { customSettings } from './config'
+import { SingleNumberResultResponse } from '@chainlink/external-adapter-framework/util'
+
+export interface CryptoRequestParams {
+  base: string
+  quote: string
+  cid?: string
+  slug?: string
+}
+
+export const inputParameters = {
+  base: {
+    aliases: ['from', 'coin', 'sym', 'symbol'],
+    description: 'The symbol of symbols of the currency to query',
+    required: true,
+    type: 'string',
+  },
+  quote: {
+    aliases: ['to', 'market', 'convert'],
+    description: 'The symbol of the currency to convert to',
+    required: true,
+    type: 'string',
+  },
+  cid: {
+    description: 'The CMC coin ID (optional to use in place of base)',
+    required: false,
+    type: 'string',
+  },
+  slug: {
+    description: 'The CMC coin ID (optional to use in place of base)',
+    required: false,
+    type: 'string',
+  },
+} as const
+
+export interface ProviderRequestBody {
+  ids: string
+  vs_currencies: string
+  include_market_cap?: boolean
+  include_24hr_vol?: boolean
+}
+
+interface PriceInfo {
+  price: number
+  volume_24h: number
+  percent_change_1h: number
+  percent_change_24h: number
+  percent_change_7d: number
+  percent_change_30d: number
+  market_cap: number
+}
+
+export interface ProviderResponseBody {
+  data: {
+    [key: string]: {
+      id: number
+      name: string
+      symbol: string
+      slug: string
+      is_active: number
+      is_fiat: number
+      circulating_supply: number
+      total_supply: number
+      max_supply: number
+      date_added: string
+      num_market_pairs: number
+      cmc_rank: number
+      last_updated: string
+      tags: string[]
+      platform: string
+      quote: {
+        [key: string]: PriceInfo
+      }
+    }
+  }
+  status: {
+    timestamp: string
+    error_code: number
+    error_message: string
+    elapsed: number
+    credit_count: number
+  }
+  cost: number
+}
+
+export type CryptoEndpointTypes = {
+  Request: {
+    Params: CryptoRequestParams
+  }
+  Response: SingleNumberResultResponse
+  CustomSettings: typeof customSettings
+  Provider: {
+    RequestBody: ProviderRequestBody
+    ResponseBody: ProviderResponseBody
+  }
+}
+
+const groupByBaseOptions = (params: CryptoRequestParams[]) => {
```

Comment:
I think theres some missing logic from v2 that we can add here. We maintain a list of cid that we use to convert symbols if it's in the map. https://github.com/smartcontractkit/external-adapters-js/blob/d0adc1082a233b695b58bf23d752506d85ca1fd4/packages/sources/coinmarketcap/src/endpoint/crypto.ts#L66-L102

Reply:
Also when you add this map to v3, you may want to update OHM to `OHM: 16209`. I think there's a PR open to update this in v2